### PR TITLE
Implement stateful class for dct

### LINKED ISSUE #17431: dct

### PR DESCRIPTION
**Pull request by rubada**
Close #17431

### CODE REVIEW

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@rubada Thanks for making the PR! Looks like the test you wrote is failing, could you try to get it passing. You can use the command pytest ivy_tests/test_ivy/test_stateful/test_layers.py::test_dct to run the test.

**Comment by rubada** [comment]
@Sam-Armstrong Hi Can you review my code.
Thanks

File: ivy/stateful/layers.py [hunk 1]
```diff
@@ -1983,3 +1983,66 @@ def _forward(self, inputs):
...
+
+class Dct(Module):
+    def __init__(
+        self,
+        *,
+        type=2,
+        n=None,
+        axis=-1,
+        norm=None,
+        out=None,
```
Comment (Sam-Armstrong) [inline ivy/stateful/layers.py:1996]: We don't need this 'out' kwarg in stateful layers. Please can you remove it, its description in the docstring, etc?

File: ivy/stateful/layers.py [hunk 2]
```diff
@@ -1983,3 +1983,66 @@ def _forward(self, inputs):
...
+        Class for applying the Discrete Cosine Transform over mini-batch of inputs.
+
+        Parameters
+        ----------
+        x
+            The input signal.
+        type
+            The type of the dct. Must be 1, 2, 3 or 4.
+        n
+            The lenght of the transform. If n is less than the input signal lenght,
```
Comment (Sam-Armstrong) [inline ivy/stateful/layers.py:2010]: typo in the word 'length'

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@rubada sorry for the delay! I've left a couple of comments for minor things to change, but this generally looks really good, so we should be good to merge once that's done. Thanks 🙂

File: ivy/stateful/layers.py (hunk 2 above)
Comment (rubada) [reply ivy/stateful/layers.py]: Thanks🙏🏻

**Review by Sam-Armstrong — APPROVED** [review]
lgtm! thanks @rubada 😁

**Comment by shephinphilip** [comment]
I am encountering error 

 pip install --target=C:/Users/USER/AppData/Local/Programs/Python/Python311 --user -e .
ERROR: Can not combine '--user' and '--target'

and another error 

from ivy.functional.ivy.layers import _deconv_length
ModuleNotFoundError: No module named 'ivy.functional'

please help me out in this

**Pull request merged**